Open-Rival/src/Rival.cpp

```diff
@@ -470,11 +470,22 @@ namespace Rival {
             dy *= -1;
         }
 
+        int mouseX;
+        int mouseY;
+        SDL_GetMouseState(&mouseX, &mouseY);
+        const float mouseXwu(RenderUtils::pxToWorld_X(mouseX));
+        const float mouseYwu(RenderUtils::pxToWorld_Y(mouseY));
+        const float midXwu(RenderUtils::pxToWorld_X(0.5 * windowWidth));
+        const float midYwu(RenderUtils::pxToWorld_Y(0.5 * windowHeight));
+        const float vecXwu(mouseXwu - midXwu);
+        const float vecYwu(mouseYwu - midYwu);
+
         if (dy > 0) {
             camera->modZoom(Camera::zoomInterval);
         } else if (dy < 0) {
             camera->modZoom(-Camera::zoomInterval);
         }
+        camera->translate(0.25 * vecXwu, 0.25 * vecYwu);
```

Comment:
Would you mind adding a comment or two here to explain the logic?

Reply:
I left out comments since I am not good with wording and not sure how these -exactly- work. We get mouse position (whatever SDL gives), convert it into some "world units", we take the middle of the screen (which I couldn't do with camera->getX/Y btw), construct a vector from midpoint to mouse position, and translate camera by that vector with a "nice quotient" :D

Reply:
Ok, thanks for the explanation, I may come back and annotate this later.

Reply:
Would really appreciate it. I'm not good with words even in my own language :P